### PR TITLE
Add IonQuant recipe.

### PR DESCRIPTION
**Pull request by reid-wagner**
Adds the IonQuant application: http://ionquant.nesvilab.org/.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by reid-wagner** [comment]
@BiocondaBot please add label

**Comment by reid-wagner** [comment]
@nsoranzo - I made the same changes here as those requested in #45127.

File: recipes/ionquant/ionquant.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import sys
+import subprocess
+from os import access, getenv, X_OK
+
+jar_file = os.path.join(os.path.dirname(__file__), "IonQuant.jar")
+
+default_jvm_mem_opts = ['-Xms512m', '-Xmx1g']
+
+license_agreement_text = '''
+Please provide pass a license key with the --key argument. You may obtain a key by agreeing to the terms at msfragger-upgrader.nesvilab.org/upgrader/.
```
Comment (bernt-matthias) [inline recipes/ionquant/ionquant.py:17]: Is this the same URL .. i.e. same key?

Comment (reid-wagner) [reply recipes/ionquant/ionquant.py]: No, thanks for catching that!

**Comment by reid-wagner** [comment]
Hi @bernt-matthias - is there anything else needed for this PR?

File: recipes/ionquant/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  summary: A label free quantification tool
+  description: |
+    IonQuant is a fast and comprehensive tool for MS1 precursor intensity-based quantification for timsTOF PASEF DDA and non-timsTOF (e.g., Orbitrap) data.
+    It enables label-free quantification with false discovery (FDR) controlled match-between-runs (MBR).
+    It can also be used for quantification in labelling-based experiments such as those involving SILAC, dimethyl, or similar labelling strategies.
+
+    IonQuant is available freely for academic research and educational purposes only, in accordance with the terms at https://msfragger.arsci.com/ionquant/IonQuant%20Academic%20Use%20License%2005162022.pdf.
+
+extra:
+  notes: |
```
Comment (bernt-matthias) [inline recipes/ionquant/meta.yaml:44]: Want to add yourself as recipe maintainer?

```suggestion
  recipe-maintainers:
    - reid-wagner
  notes: |
```

**Review by bernt-matthias — APPROVED** [review]

**Review by bernt-matthias — APPROVED** [review]

**Comment by reid-wagner** [comment]
Thank you!

**Pull request merged**